### PR TITLE
Backport critical fixes from develop

### PR DESCRIPTION
**Pull request by UdjinM6**
Things to include:
- [x] #1810 
- [x] #1819 
- [x] bp#8256 (3f7273d from #1821)
- [x] #1826 
- [x] #1829 
- [x] #1833 
- [x] #1834 
- [x] #1836
- [x] #1840 

Note: this should be merged without squashing

### CODE REVIEW

**Comment by UdjinM6** [comment]
Rebased and added 3f7273d00 (BUG: bitcoin-qt crash) from #1821

**Comment by UdjinM6** [comment]
Updated PR description (added TODO list) to make it easier to follow the progress.

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -132,7 +132,8 @@ OverviewPage::OverviewPage(const PlatformStyle *platformStyle, QWidget *parent)
     currentWatchOnlyBalance(-1),
     currentWatchUnconfBalance(-1),
     currentWatchImmatureBalance(-1),
-    txdelegate(new TxViewDelegate(platformStyle, this))
+    txdelegate(new TxViewDelegate(platformStyle, this)),
+    timer(nullptr)
```
Comment (willwray) [inline src/qt/overviewpage.cpp]: `timer` should be initialised first in the list.
gcc 7.1 gives this warning:
```
In file included from ../../src/qt/overviewpage.cpp:6:0:
../../src/qt/overviewpage.h: In constructor ‘OverviewPage::OverviewPage(const PlatformStyle*, QWidget*)’:
../../src/qt/overviewpage.h:64:21: warning: ‘OverviewPage::txdelegate’ will be initialized after [-Wreorder]
     TxViewDelegate *txdelegate;
                     ^~~~~~~~~~
../../src/qt/overviewpage.h:50:13: warning:   ‘QTimer* OverviewPage::timer’ [-Wreorder]
     QTimer *timer;
             ^~~~~
../../src/qt/overviewpage.cpp:124:1: warning:   when initialized here [-Wreorder]
 OverviewPage::OverviewPage(const PlatformStyle *platformStyle, QWidget *parent) :
 ^~~~~~~~~~~~
```

Comment (UdjinM6) [reply src/qt/overviewpage.cpp]: Yep, noticed this as well. However, a) this is not critical imo and b) should be fixed in `develop`, not here.

**Comment by UdjinM6** [comment]
Added #1826 and #1829

**Comment by UdjinM6** [comment]
Added #1836

**Comment by bodapanxu** [comment]
您好，可以合作开发一个山寨DASH吗  emial：360960690@qq.com

**Comment by UdjinM6** [comment]
Added #1833 and #1834 

This should be ready for the final review now.

**Comment by codablock** [comment]
Please see https://github.com/dashpay/dash/pull/1834/files#r160357936 for a possible locking problem.

**Comment by UdjinM6** [comment]
Added #1840

**Pull request merged**